Sources/StreamVideoSwiftUI/Utils/PictureInPicture/StreamPictureInPictureController.swift

```diff
@@ -0,0 +1,138 @@
+    }
+
+    public func pictureInPictureController(_ pictureInPictureController: AVPictureInPictureController, failedToStartPictureInPictureWithError error: Error) {
+        log.debug("picture in picture failed to start called \(error)")
+    }
+
+    public func pictureInPictureControllerWillStopPictureInPicture(_ pictureInPictureController: AVPictureInPictureController) {
+        log.debug("picture in picture will stop called")
+    }
+
+    public func pictureInPictureControllerDidStopPictureInPicture(_ pictureInPictureController: AVPictureInPictureController) {
+        log.debug("picture in picture did stop called")
+    }
+
+    // MARK: - Private helpers
+
+    private func didUpdate(_ track: RTCVideoTrack?) {
+        contentViewController.track = track
+    }
+
+    private func didUpdate(_ sourceView: UIView?) {
+        if let sourceView {
+            if pictureInPictureController?.isPictureInPictureActive != false {
+                pictureInPictureController = .init(
+                    contentSource: .init(
+                        activeVideoCallSourceView: sourceView,
+                        contentViewController: contentViewController
+                    ))
+                pictureInPictureController?.canStartPictureInPictureAutomaticallyFromInline = canStartPictureInPictureAutomaticallyFromInline
+                pictureInPictureController?.delegate = self
+
+                pictureInPictureController?
+                    .publisher(for: \.isPictureInPicturePossible)
+                    .removeDuplicates()
+                    .sink { log.debug("isPictureInPicturePossible:\($0)") }
+                    .store(in: &cancellableBag)
+
+                pictureInPictureController?
+                    .publisher(for: \.isPictureInPictureActive)
+                    .removeDuplicates()
+                    .sink { [weak self] in self?.didUpdatePictureInPictureActiveState($0)  }
+                    .store(in: &cancellableBag)
+            } else {
+                pictureInPictureController?.contentSource = .init(
+                    activeVideoCallSourceView: sourceView,
+                    contentViewController: contentViewController
+                )
+            }
+        } else {
+            pictureInPictureController?.contentSource = nil
+        }
+    }
+
+    private func didUpdatePictureInPictureActiveState(_ isActive: Bool) {
+        log.debug("isPictureInPictureActive:\(isActive)")
+        setUpTrackActivator()
+    }
+
+    private func setUpTrackActivator() {
+        guard
+            pictureInPictureController?.isPictureInPictureActive == true
+        else {
+            log.debug("Picture in Picture trackActivator is disabled!")
+            ensureActiveTrackIsEnabledCancellable?.cancel()
+            ensureActiveTrackIsEnabledCancellable = nil
+            return
+        }
+
+        log.debug("Picture in Picture trackActivator is active!")
+        ensureActiveTrackIsEnabledCancellable = Timer
```

Comment:
why is this needed? Does it try to overwrite something that was done in the LLC?

Reply:
This is to ensure that the Picture in Picture track will be enabled. As Picture in Picture isn't a responsibility of LLC I don't think that should be handled there. For now LLC pauses all tracks and after that doesn't have to worry about anything else.